extensions/aws/processors/FetchS3Object.h

```diff
@@ -0,0 +1,110 @@
+/**
+ * @file FetchS3Object.h
+ * FetchS3Object class declaration
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <sstream>
+#include <utility>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "S3Processor.h"
+#include "utils/GeneralUtils.h"
+
+template<typename T>
+class S3TestsFixture;
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace aws {
+namespace processors {
+
+class FetchS3Object : public S3Processor {
+ public:
+  static constexpr char const* ProcessorName = "FetchS3Object";
+
+  // Supported Properties
+  static const core::Property Version;
+  static const core::Property RequesterPays;
+
+  // Supported Relationships
+  static const core::Relationship Failure;
```

Comment:
Is there any aws processor these two relationships are not applicable to?
In case the answer is yes, they are fine here, otherwise I would move them to the base.

Reply:
The next S3 processor depending on this is the ListS3 which only has a success relationship.